components/expenses/Expense.js

```diff
@@ -500,8 +505,50 @@ class Expense extends React.Component {
           {this.state.error && (
             <MessageBox type="error" withIcon mb={2}>
               {formatErrorMessage(intl, this.state.error)}
+              <br />
+              {formatErrorMessage(intl, this.state.error).startsWith(
+                'Host has two-factor authentication enabled for large payouts.',
+              ) && (
+                <FormattedMessage
+                  id="PayExpenseModal.HostTwoFactorAuthEnabled"
+                  defaultMessage="Please go to your {settings} to enable two-factor authentication on your account."
+                  values={{
+                    settings: (
+                      <StyledLink as={Link} route="editCollective" params={{ slug: LoggedInUser.collective.slug }}>
+                        account settings
+                      </StyledLink>
+                    ),
+                  }}
+                />
+              )}
             </MessageBox>
           )}
+          {this.state.error && formatErrorMessage(intl, this.state.error).startsWith('Two-factor authentication') && (
+            <StyledInputField
+              name="twoFactorAuthenticatorCode"
+              htmlFor="twoFactorAuthenticatorCode"
+              label={
+                <FormattedMessage
+                  id="PayExpenseModal.TwoFactorAuthCode"
+                  defaultMessage="Two-factor authentication code"
+                />
+              }
+              mt={2}
+              mb={3}
+            >
+              {inputProps => (
+                <StyledInput
+                  {...inputProps}
+                  minHeight={50}
+                  fontSize="20px"
+                  placeholder="123456"
+                  pattern="[0-9]{6}"
```

Comment:
This is good :+1: 

I suggest adding the `inputMode="numeric"` prop, which will force touchscreens and accessibility tools to render a numeric keyboard (I often forget to use this prop myself). See https://developer.mozilla.org/fr/docs/Web/HTML/Attributs_universels/inputmode for more info